internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/router/TestRead.java

```diff
@@ -753,4 +836,157 @@ public void testRequestUsageMetric() {
+        for (int k = 0; k < initialKeyCount; k++) {
+          assertEquals(client.get(k).get(), i);
+        }
+        clients.add(client);
+      }
+      // Verify new push works
+      final int newValue = 1000;
+      try (ControllerClient parentControllerClient = new ControllerClient(clusterNames[0], parentControllerURLs)) {
+        PubSubProducerAdapterFactory pubSubProducerAdapterFactory = multiClusterVenice.getClusters()
+            .get(clusterNames[0])
+            .getPubSubBrokerWrapper()
+            .getPubSubClientsFactory()
+            .getProducerAdapterFactory();
+        VersionCreationResponse versionCreationResponse = TestUtils.createVersionWithBatchData(
+            parentControllerClient,
+            stores.get(0),
+            INT_KEY_SCHEMA,
+            INT_VALUE_SCHEMA,
+            IntStream.range(0, initialKeyCount).mapToObj(i -> new AbstractMap.SimpleEntry<>(i, newValue)),
+            pubSubProducerAdapterFactory,
+            additionalPubSubProperties);
+        TestUtils.waitForNonDeterministicPushCompletion(
+            versionCreationResponse.getKafkaTopic(),
+            parentControllerClient,
+            60,
+            TimeUnit.SECONDS);
+      }
+      TestUtils.waitForNonDeterministicAssertion(2, TimeUnit.MINUTES, true, () -> {
+        for (int k = 0; k < initialKeyCount; k++) {
+          assertEquals(clients.get(0).get(k).get(), newValue);
+        }
+      });
+
+      String aNewStoreName = stores.get(0);
+      VeniceClusterWrapper vcw = multiClusterVenice.getClusters().get(clusterNames[0]);
+
+      TestUtils.waitForNonDeterministicAssertion(2, TimeUnit.MINUTES, true, () -> {
+        vcw.updateStore(aNewStoreName, new UpdateStoreQueryParams().setBlobTransferEnabled(true));
+      });
+
+      String routerURL = vcw.getRandomRouterURL();
+      try (CloseableHttpAsyncClient client = HttpAsyncClients.custom()
+          .setDefaultRequestConfig(RequestConfig.custom().setSocketTimeout(2000).build())
+          .build()) {
+        client.start();
+
+        String uri = routerURL + "/" + TYPE_BLOB_DISCOVERY + "?store_name=" + aNewStoreName
+            + "&store_version=1&store_partition=1";
+        HttpGet routerRequest = new HttpGet(uri);
+        HttpResponse response = client.execute(routerRequest, null).get();
+        String responseBody;
+        try (InputStream bodyStream = response.getEntity().getContent()) {
+          responseBody = IOUtils.toString(bodyStream);
+        }
+        Assert.assertEquals(
+            response.getStatusLine().getStatusCode(),
+            HttpStatus.SC_OK,
+            "Failed to get resource state for " + storeVersionName + ". Response: " + responseBody);
+        ObjectMapper mapper = ObjectMapperFactory.getInstance();
+        BlobDiscoveryResponse blobDiscoveryResponse =
+            mapper.readValue(responseBody.getBytes(), BlobDiscoveryResponse.class);
+        Assert.assertEquals(blobDiscoveryResponse.getLiveNodeHostNames().size(), 0);
```

Comment:
Please also add a TODO here since we need to come back and improve the test. A response contains 0 node is not sufficient